system/service/auth.go

```diff
@@ -1371,3 +1425,23 @@ func (svc auth) LoadRoleMemberships(ctx context.Context, u *types.User) error {
 func (svc auth) GetProviders() types.ExternalAuthProviderSet {
 	return CurrentSettings.Auth.External.Providers
 }
+
+// randomStringWithRestrictions generates a random ASCII string with at least one digit and one special character
+func randomStringWithRestrictions() string {
```

Comment:
Could this work if we move it to `pkg/rand:Password()`? You can design it in a similar way as `rand.Bytes()`